### PR TITLE
Restart the daemon if any of the the plugin config files are modified

### PR DESCRIPTION
**Pull request by hughsie**
Fixes https://github.com/fwupd/fwupd/issues/3625

Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [X] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

**Comment by superm1** [comment]
Is it a crazy idea to instead unload and reload plugin? Given that is what happens for the other configuration files in the daemon just wondering if it makes sense rather than take down the whole daemon.

**Comment by hughsie** [comment]
> Is it a crazy idea to instead unload and reload plugin

I did ponder that, but there's just too much going on. e.g. recoldplugs, handling registration of other plugins etc -- it's much more safe to just quit when it's safe and then start up from scratch again. I don't think it's the sort of thing that's going to happen very often in reality.

File: src/fu-engine.c [hunk 1]
```diff
@@ -5761,6 +5761,13 @@ fu_engine_plugin_rules_changed_cb (FuPlugin *plugin, gpointer user_data)
 	}
 }
 
+static void
+fu_engine_plugin_config_changed_cb(FuPlugin *plugin, gpointer user_data)
+{
+	FuEngine *self = FU_ENGINE(user_data);
+	fu_engine_set_status(self, FWUPD_STATUS_SHUTDOWN);
```
Comment (superm1) [inline src/fu-engine.c:5768]: I think you need a non-verbose message here for the reason to shut down the daemon for the journal.

**Review by superm1 — DISMISSED** [review]
one nit and then looks okay

**Comment by superm1** [comment]
> > Is it a crazy idea to instead unload and reload plugin
> 
> I did ponder that, but there's just too much going on. e.g. recoldplugs, handling registration of other plugins etc -- it's much more safe to just quit when it's safe and then start up from scratch again. I don't think it's the sort of thing that's going to happen very often in reality.

very true, sounds good.

**Review by superm1 — APPROVED** [review]

**Pull request merged**